### PR TITLE
Add logout button and snapshot test

### PR DESCRIPTION
**Pull request by Gwillison415**
need to write more tests

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_GOOGLE_AUTH_URL=http://localhost:8003/auth/google
```
Comment (joshwyatt) [inline .env.development]: @Gwillison415 nice, thanks for using the `.development` suffix.

File: package.json [hunk 2]
```diff
@@ -14,19 +15,23 @@
     "semantic-ui-react": "0.71.5"
   },
   "devDependencies": {
+    "enzyme": "^2.9.1",
```
Comment (joshwyatt) [inline package.json]: @Gwillison415 if you wouldn't mind removing the `^` here and elsewhere in the dependencies, thanks!

File: src/App.jsx [hunk 3]
```diff
@@ -12,4 +13,4 @@ const App = () => (
   </div>
 );
 
-export default App;
+export default connect()(App);
```
Comment (joshwyatt) [inline src/App.jsx]: @Gwillison415 This is over my head, can you tell me why the need here to connect the app component?

File: src/App.test.jsx [hunk 4]
```diff
@@ -1,8 +1,23 @@
...
+import SHEET_API from './utils/Api';
+import App from './App.jsx';
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
+const middleware = [ thunkMiddleware.withExtraArgument({ SHEET_API })];
+const mockStore = configureStore(middleware);
+describe('App component', () => {
+  it('should render a component with props as specified ', () => {
```
Comment (joshwyatt) [inline src/App.test.jsx]: @Gwillison415 perhaps the `connected` component was used in order to write this test?

@HamidAghdaee would you mind weighing in on this one here? Thanks!

File: src/Components/Login/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable import/no-named-as-default, no-shadow, consistent-return */
+import React, { Component } from 'react';
+import { Grid, Card, Button } from 'semantic-ui-react';
+// import { connect } from 'react-redux';
+// import PropTypes from 'prop-types';
+import './Login.css';
+
+// NOTE all commented out code will be used in future mapping of state, dispatching of actions etc
```
Comment (joshwyatt) [inline src/Components/Login/Login.jsx]: @Gwillison415 helpful comment, thanks.

File: src/Components/Login/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+// import { renderSheets } from '../../actions/renderActions';
+// import injectWidgetId from '../../utils/utils';
```
Comment (joshwyatt) [inline src/Components/Login/Login.jsx]: @Gwillison415 I think it's great that you left mention of why this comment is in here, and even did some of the future work preemptively, but I'm not sure if it's a good idea to comment out imports for directories that don't even exist yet. It's easy for me to imagine future developers making different choices about their file structure, based on considerations that are not yet know to us, and therefore I invoke the principle of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), and ask you to remove the not-yet-needed comments.

As an act of kindness though you might cut an additional branch off of this `stephen` branch before making an additional commit, and notifying the rest of the team they could find some potentially helpful starter code should they begin work on the feature in the future.

File: src/actions/loginAction.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export const RENDER_LOGIN = 'RENDER_LOGIN';
+
+export const renderLogin = () => ({
+  type: RENDER_LOGIN,
+});
+
+// in component using mapDisptachToProps dispatch the above action creator
```
Comment (joshwyatt) [inline src/actions/loginAction.js]: @Gwillison415 This comment I do not believe you meant to keep in.

File: package.json (hunk 2 above)
Comment (Gwillison415) [reply package.json]: right sorry about that

File: src/Components/Login/Login.jsx (hunk 6 above)
Comment (Gwillison415) [reply src/Components/Login/Login.jsx]: YAGNI !! ok thanks for the ideas and next steps. will cut after I fix other issues

File: src/App.jsx (hunk 3 above)
Comment (Gwillison415) [reply src/App.jsx]: ``` mapStateToProps``` and``` mapDispatchToProps ``` are both pure functions that are provided the stores “state” and “dispatch” respectively.  only components within the Provider can be connected however our app.jsx is yet to change the state of the application. When we institute a login componentWillMount inside of ```Login.jsx``` the App will conditionally render based on login and need the connect function that takes mpstp and mdtp. Does this help clarify? I'm not sure if this was for testing but for linting.. 5 days ago.. what say you @HamidAghdaee ?

File: src/App.test.jsx (hunk 4 above)
Comment (HamidAghdaee) [reply src/App.test.jsx]: yeah I agree with @joshwyatt Don't test the connected component, export the unconnected one and test it.

File: src/App.jsx (hunk 3 above)
Comment (HamidAghdaee) [reply src/App.jsx]: indeed @Gwillison415 why connect here?

File: src/Components/Login/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+/* eslint-disable react/prefer-stateless-function */
+export class Login extends Component {
+  // componentDidMount() {
```
Comment (HamidAghdaee) [inline src/Components/Login/Login.jsx]: these commented out lines seem to be for future development. Why not include them here? for the sake of breaking up the functionality into a number of smaller PRs? @Gwillison415

File: src/App.jsx [hunk 9]
```diff
@@ -13,4 +13,5 @@ const App = () => (
   </div>
 );
 
-export default connect()(App);
+export default App;
+// export default connect()(App);
```
Comment (joshwyatt) [inline src/App.jsx]: Thanks @Gwillison415, but if you would please remove this comment.

File: src/Components/Login/Login.jsx [hunk 10]
```diff
@@ -1,23 +1,13 @@
...
-export class Login extends Component {
-  // componentDidMount() {
-  //   if (cookie.load('userName') !== undefined) {
-  //     const name = cookie.load('userName');
-  //     // NOTE write function to trigger the render of the main sheets page
-  //   }
-  // }
-  render() {
-    const GOOGLE_AUTH_URL = process.env.REACT_APP_GOOGLE_AUTH_URL;
+export const Login = () => {
```
Comment (joshwyatt) [inline src/Components/Login/Login.jsx]: @Gwillison415 thanks also for recognizing this is now more well suited as a functional component.

**Comment by joshwyatt** [comment]
Thanks @Gwillison415! @HamidAghdaee this looks good to me (except for that one lingering comment which I'm sure @Gwillison415 will remove promptly), would you mind giving it one final pass and merging it?

File: src/App.test.jsx [hunk 11]
```diff
@@ -1,8 +1,23 @@
...
 
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
+const middleware = [thunkMiddleware.withExtraArgument({ SHEET_API })];
+const mockStore = configureStore(middleware);
+describe('App component', () => {
+  it('should render a component with props as specified ', () => {
+    const component = shallow(
+      <Provider store={mockStore({})}>
```
Comment (joshwyatt) [inline src/App.test.jsx]: @HamidAghdaee I know we haven't (in our work here at G) tended to shallow test our components as they are wrapped in higher order components, what do you think?

File: .env.development (hunk 1 above)
Comment (HamidAghdaee) [reply .env.development]: @Gwillison415 let's just have one URL: REACT_APP_SHEETS_API, and append endpoints to it in the app. This needs to be addressed in this PR

File: src/App.test.jsx (hunk 11 above)
Comment (HamidAghdaee) [reply src/App.test.jsx]: @Gwillison415 I agree with @joshwyatt no need to test a connected component.

**Comment by HamidAghdaee** [comment]
@Gwillison415 After fixing the situation in Github, please update your code here to be in line with that. also needs a rebase off upstream/dev

File: .env.development [hunk 12]
```diff
@@ -0,0 +1 @@
+REACT_APP_GOOGLE_URL=http://localhost:8003
```
Comment (HamidAghdaee) [inline .env.development]: This should be REACT_APP_SHEETS_API_URL=http://localhost:8003

Talk to @michaelmurray6298 if  you have questions about it.

Comment (HamidAghdaee) [reply .env.development]: @Gwillison415, @michaelmurray6298 just merged the right thing for this URL, please rebase from it and use it.

**Pull request merged**